src/zh/jinmantiantang/src/eu/kanade/tachiyomi/extension/zh/jinmantiantang/JinmantiantangPreferences.kt

```diff
@@ -118,9 +118,15 @@ class UpdateUrlInterceptor(private val preferences: SharedPreferences) : Interce
     var isUpdated = false
 
     override fun intercept(chain: Interceptor.Chain): Response {
-        val request = chain.request()
+        var request = chain.request()
         if (!request.url.toString().startsWith(baseUrl)) return chain.proceed(request)
 
+        request = request.newBuilder().headers(
```

Comment:
instead of this, add referer in the source class directly

```kt
override fun headersBuilder() = super.headersBuilder()
    .add("Referer", "$baseUrl/")
```